test/calibration/test_calibrations.py

```diff
@@ -0,0 +1,1054 @@
+        self.assertEqual(self.cals.get_parameter_value("σ", (3,), "x90p"), 50)
+        self.assertEqual(self.cals.get_parameter_value("σ", (3,), "xp"), 50)
+
+    def test_channel_names(self):
+        """Check the naming of parametric control channels index1.index2.index3..."""
+        drive_0 = DriveChannel(Parameter("ch0"))
+        drive_1 = DriveChannel(Parameter("ch1"))
+        control_bad = ControlChannel(Parameter("u_chan"))
+        control_good = ControlChannel(Parameter("ch1.0"))
+
+        with pulse.build(name="good_sched") as sched_good:
+            pulse.play(Drag(160, 0.1, 40, 2), drive_0)
+            pulse.play(Drag(160, 0.1, 40, 2), drive_1)
+            pulse.play(Drag(160, 0.1, 40, 2), control_good)
+
+        with pulse.build(name="bad_sched") as sched_bad:
+            pulse.play(Drag(160, 0.1, 40, 2), drive_0)
+            pulse.play(Drag(160, 0.1, 40, 2), drive_1)
+            pulse.play(Drag(160, 0.1, 40, 2), control_bad)
+
+        self.cals.add_schedule(sched_good)
+
+        with self.assertRaises(CalibrationError):
+            self.cals.add_schedule(sched_bad)
+
+    def test_unique_parameter_names(self):
+        """Test that we cannot insert schedules in which parameter names are duplicates."""
+        with pulse.build() as sched:
+            pulse.play(Drag(160, Parameter("a"), Parameter("a"), Parameter("a")), DriveChannel(0))
+
+        with self.assertRaises(CalibrationError):
+            self.cals.add_schedule(sched)
+
+    def test_parameter_without_schedule(self):
+        """Test that we can manage parameters that are not bound to a schedule."""
+        self.cals._register_parameter(Parameter("a"), ())
+
+    def test_free_parameters(self):
+        """Test that we can get a schedule with a free parameter."""
+        xp = self.cals.get_schedule("xp", 3, assign_params={"amp": self.amp_xp})
+        self.assertEqual(xp.parameters, {self.amp_xp})
+
+        xp = self.cals.get_schedule("xp", 3, assign_params={"amp": self.amp_xp, "σ": self.sigma})
+        self.assertEqual(xp.parameters, {self.amp_xp, self.sigma})
+
+    def test_qubit_input(self):
+        """Test the qubit input."""
+
+        xp = self.cals.get_schedule("xp", 3)
+        self.assertEqual(xp.instructions[0][1].operands[0].amp, 0.2)
+
+        val = self.cals.get_parameter_value("amp", 3, "xp")
+        self.assertEqual(val, 0.2)
+
+        val = self.cals.get_parameter_value("amp", (3,), "xp")
+        self.assertEqual(val, 0.2)
+
+        with self.assertRaises(CalibrationError):
+            self.cals.get_parameter_value("amp", ("3",), "xp")
+
+        val = self.cals.get_parameter_value("amp", "3", "xp")
+        self.assertEqual(val, 0.2)
+
+        with self.assertRaises(CalibrationError):
+            self.cals.get_parameter_value("amp", "(1, a)", "xp")
+
+        with self.assertRaises(CalibrationError):
+            self.cals.get_parameter_value("amp", [3], "xp")
+
+
+class TestCalibrationDefaults(QiskitTestCase):
```

Comment:
What do you think about `TestOverrideEntries`? Having calibration defaults and calibration basis are quite confusing, i.e. both can have the same meaning.

Reply:
Sounds good. See here: abbe216